File: ExtendedDataGridView Control/[C#]-ExtendedDataGridView Control/C#/ExtendedDatagridViewCS/DataGridViewColumns/altDataGridViewCheckBoxColumn.cs
using System.Windows.Forms;
using System.ComponentModel;
namespace ExtendedDataGridView
{

    public class altDataGridViewCheckBoxColumn : DataGridViewCheckBoxColumn
	{

		private enumerations.style2 _HeaderStyle;
		[Category("Design"), DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
		public enumerations.style2 HeaderStyle {
			get { return _HeaderStyle; }
			set {
				_HeaderStyle = value;
				switch (_HeaderStyle) {
					case enumerations.style2.Standard:
						base.HeaderCell = new DataGridViewColumnHeaderCell();
						break;
					case enumerations.style2.CheckAll:
						base.HeaderCell = new checkAllHeaderCell();
						break;
					case enumerations.style2.HideColumn:
						base.HeaderCell = new checkHideColumnHeaderCell();
						break;
				}
			}
		}

		public override object Clone()
		{
			altDataGridViewCheckBoxColumn copyColumn = (altDataGridViewCheckBoxColumn)base.Clone();
			copyColumn._HeaderStyle = this.HeaderStyle;
			return copyColumn;
		}

	}
}

File: ExtendedDataGridView Control/[VB]-ExtendedDataGridView Control/VB/ExtendedDataGridView/ExtendedDataGridView/DataGridViewColumns/altDataGridViewCheckBoxColumn.vb
Imports System.ComponentModel

Public Class altDataGridViewCheckBoxColumn
    Inherits DataGridViewCheckBoxColumn

    Private _HeaderStyle As enumerations.style2
    <Category("Design"), DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)>
    Public Property HeaderStyle() As enumerations.style2
        Get
            Return _HeaderStyle
        End Get
        Set(ByVal value As enumerations.style2)
            _HeaderStyle = value
            Select Case _HeaderStyle
                Case style2.Standard
                    MyBase.HeaderCell = New DataGridViewColumnHeaderCell
                Case style2.CheckAll
                    MyBase.HeaderCell = New checkAllHeaderCell
                Case style2.HideColumn
                    MyBase.HeaderCell = New checkHideColumnHeaderCell
            End Select
        End Set
    End Property

    Public Overrides Function clone() As Object
        Dim copyColumn As altDataGridViewCheckBoxColumn = DirectCast(MyBase.Clone, altDataGridViewCheckBoxColumn)
        copyColumn._HeaderStyle = Me.HeaderStyle
        Return copyColumn
    End Function

End Class
